101-vm-with-data-management-gateway/azuredeploy.json

```diff
@@ -0,0 +1,297 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "existingDataFactoryName": {
+      "type":"string",
+      "metadata": {
+        "description": "your existing data factory name"
+      }
+    },
+    "GatewayName": {
+      "type": "string",
+      "metadata": {
+        "description": "Gateway name must be unique in subscription"
+      }
+    },
+    "virtualMachineName": {
+      "type": "string",
+      "maxLength": 15,
+      "minLength": 3,
+      "metadata": {
+        "description": "Input the virtual machine name"
+      }
+    },        
+    "Size": {
+        "type": "string",
+        "defaultValue": "Standard_DS3_v2",
+        "allowedValues": ["Standard_A1_v2", "Standard_A2_v2", "Standard_A4_v2", "Standard_A8_v2", "Standard_A2M_v2", "Standard_A4M_v2", "Standard_A8M_v2",
+          "Standard_D1_v2", "Standard_D2_v2", "Standard_DS_v2", "Standard_D4_v2", "Standard_D5_v2", "Standard_DS1_v2", "Standard_DS2_v2", "Standard_DS3_v2", "Standard_DS4_v2", "Standard_DS5_v2", 
+          "Standard_A0", "Standard_A1", "Standard_A2", "Standard_A3", "Standard_A4", "Standard_A5", "Standard_A6", "Standard_A7",
+          "Standard_F1", "Standard_F2", "Standard_F4", "Standard_F8", "Standard_F16", "Standard_F1s", "Standard_F2s", "Standard_F4s", "Standard_F8s", "Standard_F16s",
+          "Standard_D11_v2", "Standard_D12_v2", "Standard_D13_v2", "Standard_D14_v2", "Standard_D15_v2",  "Standard_DS11_v2", "Standard_DS12_v2", "Standard_DS13_v2", "Standard_DS14_v2", "Standard_DS15_v2",
+          "Standard_D11", "Standard_D12", "Standard_D13", "Standard_D14",  "Standard_DS11", "Standard_DS12", "Standard_DS13", "Standard_DS14"
+        ],
+        "metadata":{
+          "description": "The size of virtual machine. Pricing link: https://azure.microsoft.com/en-us/pricing/details/virtual-machines/windows/ "
+        }
+    },
+    "adminUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for the virtual machine"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the virtual machine"
+      }
+    },
+    "existingVirtualNetworkName":{
+      "type": "string",
+      "metadata": {
+        "description": "your existing vnet name"
+      }
+    },
+    "existingVnetLocation":
```

Comment:
VM and VNET must be in the same location, otherwise the deployment will be failed. I have tried reference() function, but the output doesn't contain the location.
{
    "provisioningState": "Succeeded",
	"resourceGuid": 
	"d1c60435-1179-4931-93f5-b127d93533b8",
	"addressSpace": {
		"addressPrefixes": [
			"10.1.0.0/24"
		]
	},
	"subnets": [
	    {
	        "name": "default",
	        "id": "/subscriptions/b371d9e7-d3c2-4b1a-83ec-84e1f50c2222/resourceGroups/xiaoyztest/providers/Microsoft.Network/virtualNetworks/testvnet/subnets/default",
	        "etag": "W/\"fd4872ab-95ea-455b-ac02-65bb59fb3fd0\"",
	        "properties": {
	            "provisioningState": "Succeeded",
	            "addressPrefix": "10.1.0.0/24"
	        }
	    }
	]
}